src/otx/v2/adapters/torch/mmengine/mmaction/engine.py

```diff
@@ -0,0 +1,290 @@
+            codebase_config = {"type": codebase, "task": task}
+            deploy_config_dict["codebase_config"] = codebase_config
+
+        # if_config update
+        if ir_config is None:
+            ir_config = {
+                "type": "onnx",
+                "export_params": True,
+                "keep_initializers_as_inputs": False,
+                "opset_version": 11,
+                "save_file": "end2end.onnx",
+                "input_names": ["input"],
+                "output_names": ["output"],
+                "input_shape": None,
+                "optimize": False,
+                "dynamic_axes": {
+                    "input": {
+                        0: "batch",
+                        1: "num_crops * num_segs",
+                        3: "time",
+                        4: "height",
+                        5: "width",
+                    },
+                    "output": {
+                        0: "batch",
+                    },
+                },
+            }
+            deploy_config_dict["ir_config"] = ir_config
+        # BACKEND_CONFIG Update
+        if backend_config is None:
+            backend_config = {
+                "type": "openvino",
+                "model_inputs": [
+                    {
+                        "opt_shapes":{
+                            "input": [1, 1, 3, 32, 224, 224],
+                        },
+                    },
+                ],
+                "mo_options":{
+                    "args":{
+                        "--source_layout": "?bctwh",
+                    },
+                },
+            }
+            deploy_config_dict["backend_config"] = backend_config
+
+        # Patch input's configuration
+        if isinstance(deploy_config_dict, dict):
+            deploy_config_dict = Config(deploy_config_dict)
+        data_preprocessor = self.dumped_config.model.get("data_preprocessor", None)
+        mean = data_preprocessor["mean"] if data_preprocessor is not None else [123.675, 116.28, 103.53]
+        std = data_preprocessor["std"] if data_preprocessor is not None else [58.395, 57.12, 57.375]
+        to_rgb = data_preprocessor["to_rgb"] if data_preprocessor is not None else False
+        patch_input_preprocessing(deploy_cfg=deploy_config_dict, mean=mean, std=std, to_rgb=to_rgb)
+        if not deploy_config_dict.backend_config.get("model_inputs", []):
+            if input_shape is None:
+                pass
+            patch_input_shape(deploy_config_dict, input_shape=input_shape)
+
+        export_dir = Path(self.work_dir) / f"{self.timestamp}_export"
+        exporter = Exporter(
+            config=self.dumped_config,
+            checkpoint=str(checkpoint),
+            deploy_config=deploy_config_dict,
+            work_dir=str(export_dir),
+            precision=precision,
+            export_type=export_type,
+            device=device,
+        )
+
+        return exporter.export()
```

Comment:
It seems there are lots of duplicated codes with base engine's export function. Could you check it?

Reply:
Thanks, done. https://github.com/openvinotoolkit/training_extensions/pull/2627/commits/737b9720bc246e8928fdc805746b11b963527054